includes/collection/class-actors.php

```diff
@@ -389,4 +389,255 @@ public static function get_type_by_id( $user_id ) {
+	 */
+	public static function add( $actor_data ) {
+		if ( \is_array( $actor_data ) ) {
+			$actor_data = Actor::init_from_array( $actor_data );
+		}
+
+		if ( ! $actor_data instanceof Actor ) {
+			return new \WP_Error(
+				'activitypub_invalid_actor_data',
+				\__( 'Invalid actor data', 'activitypub' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$inbox = null;
+		if ( ! empty( $actor_data->get_endpoints()['sharedInbox'] ) ) {
+			$inbox = $actor_data->get_endpoints()['sharedInbox'];
+		} elseif ( ! empty( $actor_data->get_inbox() ) ) {
+			$inbox = $actor_data->get_inbox();
+		}
+
+		$args = array(
+			'guid'         => \esc_url_raw( $actor_data->get_id() ),
+			'post_title'   => \wp_strip_all_tags( \sanitize_text_field( $actor_data->get_name() ?? $actor_data->get_preferred_username() ) ),
+			'post_author'  => 0,
+			'post_type'    => self::POST_TYPE,
+			'post_name'    => \esc_url_raw( $actor_data->get_id() ),
+			'post_content' => \wp_slash( $actor_data->to_json() ),
+			'post_excerpt' => \sanitize_text_field( \wp_kses( \wp_slash( $actor_data->get_summary() ), 'user_description' ) ),
+			'post_status'  => 'publish',
+			'meta_input'   => array(
+				'_activitypub_inbox' => $inbox,
+			),
+		);
+
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		$post_id = $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT ID FROM $wpdb->posts WHERE guid=%s",
+				esc_sql( $actor_data->get_id() )
+			)
+		);
+
+		if ( $post_id ) {
+			// If this is an update, prevent the "followed" date from being overwritten by the current date.
+			$post                  = \get_post( $post_id );
+			$args['ID']            = $post_id;
+			$args['post_date']     = $post->post_date;
+			$args['post_date_gmt'] = $post->post_date_gmt;
+		}
+
+		$has_kses = false !== \has_filter( 'content_save_pre', 'wp_filter_post_kses' );
+		if ( $has_kses ) {
+			// Prevent KSES from corrupting JSON in post_content.
+			\kses_remove_filters();
+		}
+
+		if ( $post_id ) {
+			$post_id = \wp_update_post( $args );
+		} else {
+			$post_id = \wp_insert_post( $args );
+		}
+
+		if ( $has_kses ) {
+			// Restore KSES filters.
+			\kses_init_filters();
+		}
+
+		return $post_id;
+	}
+
+	/**
+	 * Get a remote Actor object by actor URL (guid).
+	 *
+	 * @param string $actor_uri The actor URI.
+	 *
+	 * @return \WP_Post|\WP_Error
+	 */
+	public static function get_remote_by_uri( $actor_uri ) {
+		global $wpdb;
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+		$post_id = $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT ID FROM $wpdb->posts WHERE guid=%s AND post_type=%s",
+				esc_sql( $actor_uri ),
+				esc_sql( self::POST_TYPE )
+			)
+		);
+
+		if ( ! $post_id ) {
+			return null;
+		}
+
+		return \get_post( $post_id );
+	}
+
+	/**
+	 * Get a remote Actor object by actor URL (guid).
+	 *
+	 * @param int $actor_id The actor ID.
+	 *
+	 * @return \WP_Post|\WP_Error
```

Comment:
```suggestion
	 * @return \WP_Post
```
Doesn't look like there's a WP_Error here